Close correct tab when close window called

marcosholgado opened the pull request:
<!--
Note: This checklist is a reminder of our shared engineering expectations. Feel free to change it, although assigning a GitHub reviewer and the items in bold are required.
-->

Task/Issue URL: https://app.asana.com/0/1125189844152671/1199674134590488
Tech Design URL: 
CC: 

**Description**:
This PR fixes a bug where we close the wrong tab after receiving a close window event.

**Steps to test this PR**:

**Google Sign in**
1. Install the app
1. Open a tab and go to SERP
1. Open another tab and go to a site with a google login like `etsy.com`
1. Sign in with your google account, once the flow has finished you should be returned to the tab which initiated the flow
1. Some sites require us to enable 3p cookies for the login with Google to work, in those cases you will not be signed in but you should still be returned to the correct tab.

**Facebook Sign in**
1. Install the app
1. Open a tab and go to SERP
1. Open another tab and go to a site with a Facebook login like `pinterest.com`
1. Sign in with your facebook account, once the flow has finished you should be returned to the tab which initiated the flow.


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

cmonfortep commented on app/src/main/java/com/duckduckgo/app/tabs/model/TabDataRepository.kt:
```diff
@@ -216,9 +216,9 @@ class TabDataRepository @Inject constructor(
     }
 
-    override suspend fun deleteCurrentTabAndSelectSource() {
+    override suspend fun deleteCurrentTabAndSelectSource(tabId: String) {
```
@marcosholgado I think we talked about changing this method name to remove the part of `CurrentTab` now that we are passing `tabId` as parameter. Are you ok with that?

marcosholgado replied:
👍

cmonfortep commented on app/src/main/java/com/duckduckgo/app/tabs/model/TabRepository.kt:
```diff
@@ -66,7 +66,7 @@ interface TabRepository {
     suspend fun purgeDeletableTabs()
 
-    suspend fun deleteCurrentTabAndSelectSource()
+    suspend fun deleteCurrentTabAndSelectSource(tabId: String)
```
Same here, I would remove the `currentTab` reference from the method name since we can pass any tab id as parameter.

marcosholgado replied:
👍

cmonfortep reviewed (approved):
Just a comment related to method naming. 

Tested all the scenarios posted in the description and works as described.

Throwing a question here:
- Last time we tried to fix this issue we introduced a bug where the root cause was overriding the loadURL on facebook sites. I guess that code is not present anymore...so it's not possible to reproduce it again. What will happen if a similar code is introduced again? Should we test how we will react to a similar scenario?

marcosholgado commented:
> Just a comment related to method naming.
> 
> Tested all the scenarios posted in the description and works as described.
> 
> Throwing a question here:
> 
> * Last time we tried to fix this issue we introduced a bug where the root cause was overriding the loadURL on facebook sites. I guess that code is not present anymore...so it's not possible to reproduce it again. What will happen if a similar code is introduced again? Should we test how we will react to a similar scenario?

Thanks @cmonfortep. 

We are still reloading the URL when changing the user agent so we can reproduce. Currently if you sign in with Facebook the onCloseWindow method still gets called several times so that's why I added the test case specific to Facebook.
